python/ray/serve/_private/replica.py

```diff
@@ -561,21 +561,16 @@ async def invoke_single(
             runner_method = self.get_runner_method(request_item)
             method_to_call = sync_to_async(runner_method)
             result = None
-            if len(inspect.signature(runner_method).parameters) > 0:
-                result = await method_to_call(*args, **kwargs)
-            else:
-                # When access via http http_arg_is_pickled with no args:
-                # args = (<starlette.requests.Request object at 0x7fe900694cc0>,)
-                # When access via python with no args:
-                # args = ()
-                if len(args) == 1 and isinstance(args[0], starlette.requests.Request):
```

Comment:
matching on `Request` won't work anymore once the websocket PR is in (we don't always pass `Request` for HTTP handlers).